repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -1722,4 +1736,64 @@ private boolean hasActiveLineageDirection(AtlasVertex assetVertex, AtlasEdge cur
                     return Boolean.TRUE.equals(edge.get(HAS_LINEAGE));
                 });
     }
+
+    private void cleanupEntityAttributeReferences(AtlasVertex vertex, String typeName, String attributeName) {
+        try {
+            if (isAssetType(vertex)) {
+                String guid = GraphHelper.getGuid(vertex);
+                int attributeRefsRemoved = 0;
+
+                try {
+                    List<AtlasEntityHeader> entities = fetchEntitiesUsingIndexSearch(typeName, attributeName, guid);
+
+                    for (AtlasEntityHeader entity: entities) {
+                        String entityGuid = entity.getGuid();
+                        AtlasVertex entityVertex = entityRetriever.getEntityVertex(entityGuid);
+
+                        AtlasGraphUtilsV2.removeItemFromListPropertyValue(
```

Comment:
It is possible to have same Product as an inputPort as well as output port for an Asset

Meaning an asset have same product as both input Daap & output Daap? If this is allowed via API, then you need to call `removeItemFromListPropertyValue` for both attributes for such assets. 

Ignore if this if not possible if we have any validation